src/dali_backend.cc

```diff
@@ -234,39 +232,61 @@ class DaliModelInstance : public ::triton::backend::BackendModelInstance {
                  "Failed reporting batch statistics.");
   }
 
-  /** Run inference for a given \p request and prepare a response. */
-  RequestMeta ProcessRequest(TritonResponseView response, TritonRequestView request) {
-    RequestMeta ret;
-
-    auto dali_inputs = GenerateInputs(request);
-    ret.batch_size = dali_inputs[0].meta.shape.num_samples();  // Batch size is expected to be the
-                                                               // same in every input
+  /**
+   * @brief Run inference for a given \p request and prepare a response.
+   * @return computation time interval
+   */
+  ProcessingMeta ProcessRequests(const std::vector<TritonRequest>& requests,
+                                 const std::vector<TritonResponse>& responses) {
+    ProcessingMeta ret{};
+    auto inputs_info = GenerateInputs(requests);
```

Comment:
I wonder if there should be the limit of requests number we can run at once. Assume we receive 2 samples in one requests, and then 1000 in the next one. I don't think we should adjust to handle this 1000 at once, and have a hard limit not to blow the memory consumption. Like 500 at most and slit requests if needed.

Reply:
Total batch size will not exceed the `max_batch_size` that is defined in the model config.

Reply:
👍